### PR TITLE
Fix manifest validation in Tasks.Feed

### PR DESCRIPTION
**Pull request by JohnTortugo**
Closes: https://github.com/dotnet/arcade/issues/2201

The `Location` attribute is something from the old format of the Build Manifest. Moving forward we don't need that attribute.

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/BuildManifestUtil.cs [hunk 1]
```diff
@@ -119,10 +130,15 @@ public static void WriteAsXml(this BuildModel buildModel, string filePath, TaskL
             TaskLoggingHelper log)
         {
             var attributes = MSBuildListSplitter.GetNamedProperties(manifestBuildData);
-            if(!ManifestBuildDataHasLocationProperty(attributes))
+            if(!ValidateManifestBuildData(attributes, out List<string> errors))
```
Comment (riarenas) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/BuildManifestUtil.cs]: Any reason to keep `ManifestBuildDataHasLocationProperty`?

Comment (JohnTortugo) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/BuildManifestUtil.cs]: The old `PushToBlobFeed` task uses a flow that reaches that validation.

Comment (riarenas) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/BuildManifestUtil.cs]: Makes sense

**Comment by JohnTortugo** [comment]
> Have we verified that all repos are creating their manifests with the new tasks? Failing manifest validation this late in the game due to newer required fields seems a little scary.

All repos that are in YAML-based publishing are using the new Manifest format or they would be failing to publish. I'll wait a few more days for the repos that aren't on-board to get updated.

**Comment by riarenas** [comment]
I mention it because for example, just yesterday we found out Core-Setup had its own manifest creation logic that was not setting the InitialAssetsLocation field. I'm not sure if that's still the case for the runtime repo, or repos like core-sdk that also had some custom publishing logic implemented.

**Comment by JohnTortugo** [comment]
@riarenas makes sense. I added a parameter to control whether or not validation should be performed. The validation is only performed if the manifest was created using `PushToAzureDevOpsArtifacts`.

**Pull request closed without merging**